packages/main/src/TableVirtualizer.ts

```diff
@@ -0,0 +1,279 @@
+	last: number,
+};
+
+/**
+ * @class
+ *
+ * ### Overview
+ *
+ * The `ui5-table-virtualizer` component is used inside the `ui5-table` to virtualize the table rows.
+ * It is responsible for rendering only the rows that are visible in the viewport and updating them on scroll.
+ * This allows large numbers of rows to exist, but maintain high performance by only paying the cost for those that are currently visible.
+ *
+ * <b>Note:</b> The maximum number of virtualized rows is limited by browser constraints, specifically the maximum supported height for a DOM element.
+ *
+ * ### ES6 Module Import
+ * `import "@ui5/webcomponents/dist/TableVirtualizer.js";`
+ *
+ * @constructor
+ * @extends UI5Element
+ * @public
+ * @experimental This component is not intended to be used in a productive enviroment. The API is under development and may be changed in the future.
+ */
+@customElement({ tag: "ui5-table-virtualizer" })
+
+/**
+ * Fired when the virtualizer is changed by user interaction e.g. on scrolling.
+ *
+ * @public
+ */
+@event<RangeChangeEventDetail>("range-change", {
+	detail: {
+		/**
+		 * The 0-based index of the first children currently rendered.
+		 * @public
+		 */
+		first: { type: Number },
+		/**
+		 * The 0-based index of the last children currently rendered.
+		 * @public
+		 */
+		last: { type: Number },
+	},
+})
+
+class TableVirtualizer extends UI5Element implements ITableFeature {
+	/**
+	 * Defines the height of the rows in the table.
+	 *
+	 * <b>Note:</b> This property is mandatory for the virtualization to work properly.
+	 *
+	 * @default 45
+	 * @public
+	 */
+	@property({ type: Number })
+	rowHeight = 45;
+
+	/**
+	 * Defines the total count of rows in the table.
+	 *
+	 * <b>Note:</b> This property is mandatory for the virtualization to work properly.
+	 *
+	 * @default 100
+	 * @public
+	 */
+	@property({ type: Number })
+	rowCount = 100;
+
+	/**
+	 * Defines the count of extra rows to be rendered in the table.
+	 *
+	 * @default 0
+	 * @public
+	 */
+	@property({ type: Number })
+	extraRows = 0;
+
+	readonly identifier = "TableVirtualizer";
+
+	_table?: Table;
+	_lastRowPosition: number = 0;
+	_firstRowPosition: number = 0;
+	_visibleRowCount: number = 0;
+	_tabBlockingState: TabBlocking = TabBlocking.None;
+	_onRowInvalidateBound: (invalidationInfo: InvalidationInfo) => void;
+	_onScrollBound: () => void;
+
+	constructor() {
+		super();
+		this._onScrollBound = throttle(this._onScroll.bind(this));
+		this._onRowInvalidateBound = this._onRowInvalidate.bind(this);
+	}
+
+	onTableActivate(table: Table): void {
+		this._table = table;
+		this._updateRowsHeight();
+		this._scrollContainer.addEventListener("scroll", this._onScrollBound, { passive: true });
+		this._onScroll();
+	}
+
+	onAfterRendering(): void {
+		this._table && this._table._invalidate++;
+	}
+
+	onTableAfterRendering(): void {
+		if (!this._table) {
+			return;
+		}
+
+		this._updateRowsHeight();
+		if (this._tabBlockingState & TabBlocking.Released) {
```

Comment:
Maybe instead of _tabBlockingState being of type Enum, it is an explicit number representing bit flags instead (0b0000) and you instead of defining the Enum with integer values, you use binary instead?
```ts
enum TabBlocking {
  None = 0,
  Next = 1,
  Previous = 0b10,
  Released = 0b100
}
```